lib/facility.rb

```diff
@@ -1,14 +1,97 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_accessor :name,
+                :address,
+                :phone,
+                :services,
+                :registered_vehicles,
+                :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(facility)
+    @name = facility[:name] || facility[:dmv_office] || facility[:office_name] ||  "No Name"
+    @address = address_creator(facility)
+    @phone = phone_creator(facility)
+    @services = services_creator(facility) 
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def address_creator(facility)
+    if facility[:address]
```

Comment:
This ifelse can be a bit unreadable cause its so long, but it works well. A potentially more readable way of doing long ifelse is via switch statements